### PR TITLE
Use v1.CSINode in the external-attacher

### LINKED ISSUE #197: Update external-attacher to use v1.CSINode
We can also remove the fallback on Node annotation at the same time. This change will require a major version bump because we would have to bump the min K8s version to 1.17 when the object was introduced.

We should make this change at least 1, if not 2 releases before the v1beta gets removed from Kubernetes.

/lifecycle frozen

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR changes to attacher to use the GA version of the CSINode object.

**Which issue(s) this PR fixes**:

Fixes #197

**Does this PR introduce a user-facing change?**:

```release-note
Use GA version of CSINode object.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,66 +8,63 @@ require (
...
-	k8s.io/csi-translation-lib v0.0.0-20191009030015-17db17aaadeb
+	k8s.io/api v0.0.0-20191122220107-b5267f2975e0
+	k8s.io/apimachinery v0.0.0-20191121175448-79c2a76c473a
+	k8s.io/client-go v0.0.0-20191122220542-ed16ecbdf3a0
+	k8s.io/csi-translation-lib v0.0.0-20191122224157-518651485193
 	k8s.io/klog v1.0.0
 )
 
-replace k8s.io/api => k8s.io/api v0.0.0-20190918195907-bd6ac527cfd2
+replace k8s.io/api => k8s.io/api v0.0.0-20191122220107-b5267f2975e0
```
Comment (pohly) [inline go.mod]: Which version of Kubernetes is that? Can you record that in the commit message?

I assume all code coming from Kubernetes is at that version?

Comment (msau42) [reply go.mod]: I vaguely remember seeing a script that can update all k8s dependencies. Was that something you wrote?

File: cmd/csi-attacher/main.go [hunk 2]
```diff
@@ -147,7 +147,7 @@ func main() {
 			pvLister := factory.Core().V1().PersistentVolumes().Lister()
 			nodeLister := factory.Core().V1().Nodes().Lister()
 			vaLister := factory.Storage().V1beta1().VolumeAttachments().Lister()
```
Comment (msau42) [inline cmd/csi-attacher/main.go]: Hm we should use v1 VolumeAttachment too.

File: go.mod (hunk 1 above)
Comment (pohly) [reply go.mod]: I took some snippet that was posted in one of the Kubernetes discussions around semantic versioning and added it to csi-release-tools, so it's now in all of our repos: https://github.com/kubernetes-csi/csi-release-tools/blob/master/go-get-kubernetes.sh

But it's impossible to tell from the result whether that script was used and thus whether the new replacements are consistent. We basically have to trust the PR submitter that they did the right thing. To reproduce it, the version given to the scripts needs to be documented, hence me question about the commit message.

Comment (bertinatto) [reply go.mod]: > Which version of Kubernetes is that? Can you record that in the commit message?

It was `v1.17.0-rc.1`, but today I updated it to `v1.17.0-rc.2` (which was released yesterday). I also updated the commit message to record both the k8s version and how the dependencies were updated (using the `go-get-kubernetes.sh` script).

File: cmd/csi-attacher/main.go (hunk 2 above)
Comment (bertinatto) [reply cmd/csi-attacher/main.go]: I can create another PR to address that...

**Comment by bertinatto** [comment]
/hold

Checking how to support older k8s releases

**Comment by jsafrane** [comment]
@bertinatto, with Kubernetes 1.19, IMO we don't need to worry about supporting v1beta1 of CSINode. We can require Kubernetes 1.17. Can you please continue with this PR?

**Comment by bertinatto** [comment]
> @bertinatto, with Kubernetes 1.19, IMO we don't need to worry about supporting v1beta1 of CSINode. We can require Kubernetes 1.17. Can you please continue with this PR?

Done, rebased PR. I think it's ready for review.

**Pull request merged**